### PR TITLE
color-palette: fix read/write txt file with ndv + get_raster_min_max fix

### PR DESCRIPTION
**Pull request by idanmiara**
## What does this PR do?

* gdal-utils/osgeo_utils/auxiliary/color_palette.py - fix read/write txt file with ndv, add some type hints
* autotest/pyscripts/test_gdal_utils.py - add test_utils_color_files_nv and fix typo in filename
* autotest/utilities/data/color_paletted_red_green_0-1-nv.txt - rename to reflect the contents
* autotest/utilities/data/colro_pseudocolor_spectral_0-100.[txt|qml] - fix filename typo colro -> color

* gdal-utils/osgeo_utils/auxiliary/color_table.py - color_table_from_color_palette to raise appropriate Exception

* gdal-utils/osgeo_utils/auxiliary/util.py - `get_raster_min_max` - use the dedicated `bnd.ComputeRasterMinMax()` function

## Tasklist

 - [ ] ADD YOUR TASKS HERE
 - [ ] Add test case(s)
 - [ ] Add documentation
 - [ ] Review
 - [ ] Adjust for comments
 - [ ] All CI builds and checks have passed

### CODE REVIEW

**Comment by idanmiara** [comment]
Hi @rouault 
I can't find a relation between the failing tests and this PR...

**Comment by rouault** [comment]
> I can't find a relation between the failing tests and this PR...

there's none, just that there was a temporary breakage in master, that is fixed now. If you rebase on top of it, that should be fine

**Comment by idanmiara** [comment]
> > I can't find a relation between the failing tests and this PR...
> 
> there's none, just that there was a temporary breakage in master, that is fixed now. If you rebase on top of it, that should be fine

Thanks :)

**Comment by idanmiara** [comment]
@rouault It seems that the Windows build broke...

**Comment by rouault** [comment]
> It seems that the Windows build broke...

unrelated random failure. I've restarted it

**Comment by idanmiara** [comment]
> > It seems that the Windows build broke...
> 
> unrelated random failure. I've restarted it

Thanks :)

File: autotest/pyscripts/test_gdal_utils.py [hunk 1]
```diff
@@ -174,6 +174,27 @@ def test_utils_color_files():
             assert cp1.pal[k] == v
 
 
+def test_utils_color_files_nv():
+    """ test color palettes with and without nv """
+    items = [
```
Comment (rouault) [inline autotest/pyscripts/test_gdal_utils.py:179]: you could use @pytest.mark.parametrize("name,ndv",[['color_paletted_red_green_0-255.txt', None], ['color_paletted_red_green_0-1-nv.txt'', 0]])

File: gdal/swig/python/gdal-utils/osgeo_utils/auxiliary/util.py [hunk 2]
```diff
@@ -312,10 +312,8 @@ def get_raster_minimum(filename_or_ds: PathOrDS, bnd_index: Optional[int] = 1):
 def get_raster_min_max(filename_or_ds: PathOrDS, bnd_index: int = 1):
     with OpenDS(filename_or_ds) as ds:
         bnd = ds.GetRasterBand(bnd_index)
-        bnd.ComputeStatistics(0)
-        min_val = bnd.GetMinimum()
-        max_val = bnd.GetMaximum()
-        return min_val, max_val
+        min_max = bnd.ComputeRasterMinMax()
```
Comment (rouault) [inline gdal/swig/python/gdal-utils/osgeo_utils/auxiliary/util.py:315]: note: by default ComputeRasterMinMax() will recompute the statistics and not use possibly stored ones. if passing approx_ok = 1, then stored ones could be used

File: autotest/pyscripts/test_gdal_utils.py (hunk 1 above)
Comment (idanmiara) [reply autotest/pyscripts/test_gdal_utils.py]: nice, thanks!

File: gdal/swig/python/gdal-utils/osgeo_utils/auxiliary/util.py (hunk 2 above)
Comment (idanmiara) [reply gdal/swig/python/gdal-utils/osgeo_utils/auxiliary/util.py]: Thanks, I've added this parameter.

**Pull request merged**